cmd/api/main.go

```diff
@@ -39,12 +39,15 @@ import (
 	"github.com/textileio/go-tableland/pkg/sqlstore/impl/user"
```

Comment:
A big chunk of what happened in this file is a step forward in the https://github.com/tablelandnetwork/go-tableland/issues/302 direction.

I believe that our `main` function is a bit convoluted, so what I did here is a semi-proposal experiment to show if the same strategy could be done iteratively in other parts.

I encapsulated all the wiring related to `telemetry` in a single function, and only call that in `main`, which:
1) Makes it clear what is needed to wire the "telemetry module". There's no need to understand how other wires are related to telemetry.
2) The function returns a `moduleClose` (L49) that should be used by `main` to gracefully close _everything_ related to telemetry. This helps `main` to have a clear rule of always knowing that it should call returned `moduleClose` functions when the process is closed. 
3) Point 2) also helps the `cli.HandleInterrupt(...)` logic be clearer. Instead of closing fine-grained moving parts, it closes "higher-order" modules (which were built by the corresponding wiring function which knows what is needed, not "main").

The result should be a shorter, high-level, and clearer `main` function. 
I applied this idea only with the `telemetry` module since it's related to this PR. Also, to see if this starts to feel convincing.

Reply:
👍 as a suggestion for later. we can move to an `App` struct that has methods for configuring the "app". And in the main, we would have something like `app.ConfigureXXX().ConfigureYYY().Run()`.